pybop/parameterisation.py

```diff
@@ -14,25 +13,48 @@ class Parameterisation:
     Parameterisation class for pybop.
     """
 
-    def __init__(self, model, observations, fit_parameters, x0=None, options=None):
-        self.model = model
+    def __init__(
+        self,
+        model,
+        observations,
+        fit_parameters,
+        geometry=None,
+        submesh_types=None,
+        var_pts=None,
+        spatial_methods=None,
+        solver=None,
+        x0=None,
+        check_model=True,
+        init_soc=None,
+    ):
+        self.model = model.pybamm_model
+        self._unprocessed_model = model.pybamm_model
         self.fit_parameters = fit_parameters
         self.observations = observations
-        self.options = options
         self.bounds = dict(
-            lower=[p.bounds[0] for p in fit_parameters],
-            upper=[p.bounds[1] for p in fit_parameters],
+            lower=[self.fit_parameters[p].bounds[0] for p in self.fit_parameters],
+            upper=[self.fit_parameters[p].bounds[1] for p in self.fit_parameters],
         )
-
-        if options is not None:
-            self.parameter_set = options.parameter_set
+        self._model_with_set_params = None
+        self._built_model = None
+        self._geometry = geometry or self.model.default_geometry
```

Comment:
again, a lot of pybamm specific stuff. Try to keep this all in a `BaseModel` class or something so if pybamm changes its api or we decide to start using other types of models, we only have to change a single class